rllib/models/experimental/catalog.py

```diff
@@ -0,0 +1,99 @@
+from ray.rllib.models.experimental.configs import (
+    MLPModelConfig,
+    MLPEncoderConfig,
+    LSTMEncoderConfig,
+)
+from ray.rllib.models.experimental.base import ModelConfig
+from ray.rllib.models.catalog import MODEL_DEFAULTS
+from gymnasium.spaces import Box
+
+
+class Catalog:
+    """Defines how what models an RLModules builds.
+
+    RLlib's native RLModules get their Models from a ModelBuilder object.
+    By default, that ModelBuilder builds the configs it holds.
+    You can modify the ModelBuilder so that it builds different Models by subclassing
+    and don't have to write configs.
+    """
+
+    def __init__(self, observation_space, action_space, model_config):
+        self.observation_space = observation_space
+        self.action_space = action_space
+        # TODO (Artur): Possibly get rid of this config merge
+        self.model_config = {**MODEL_DEFAULTS, **model_config}
+        self.encoder_config = self.get_encoder_config(
+            observation_space, self.model_config
+        )
+        self.latent_dim = self.encoder_config.output_dim
+
+    def build_encoder(self, framework="torch"):
+        return self.encoder_config.build(framework=framework)
+
+    @staticmethod
+    def get_encoder_config(observation_space, model_config) -> ModelConfig:
+        """Returns a Model config for the given observation space.
+
+        Args
+        """
+        assert (
+            len(observation_space.shape) == 1
+        ), "No multidimensional obs space supported."
+
+        activation = model_config["fcnet_activation"]
+        output_activation = model_config["fcnet_activation"]
+        input_dim = observation_space.shape[0]
+        fcnet_hiddens = model_config["fcnet_hiddens"]
+
+        if model_config["use_lstm"]:
+            encoder_config = LSTMEncoderConfig(
+                input_dim=input_dim,
+                hidden_dim=model_config["lstm_cell_size"],
+                batch_first=not model_config["_time_major"],
+                num_layers=1,
+                output_dim=model_config["lstm_cell_size"],
+                output_activation=output_activation,
+            )
+        else:
+            encoder_config = MLPEncoderConfig(
+                input_dim=input_dim,
+                hidden_layer_dims=fcnet_hiddens[:-1],
+                hidden_layer_activation=activation,
+                output_dim=fcnet_hiddens[-1],
+                output_activation=output_activation,
+            )
+        return encoder_config
+
+    @staticmethod
+    def get_base_model_config(input_space, model_config) -> ModelConfig:
```

Comment:
It's not clear how base_model_config is related to encoder. For what it matters I could say I'll just call catalog.build_encoder() right? So what's the use case for base_model_config?

Reply:
We need this to get a "standard" encoder config for the constructor.
Upon instantiation, any Catalog object should already have some ModelConfigs that users can change to enable workflows like these:

```
catalog = PPOCatalog()
catalog.encoder_config.hidden_activation = "symlog"
catalog.build_encoder()
```

That is how, in the future, I'd like users to configure models through this instead of using the model_config in the future so that we don't have two ways of configuring that can get in the way of each other.

Reply:
Rather, each model_configs should just map to a "standard initialized catalog".
But not both should be provided.